crates/oxc_linter/src/rules/eslint/getter_return.rs

```diff
@@ -426,6 +478,31 @@ fn test() {
         ),
         ("var foo = { get bar() { try { return a(); } catch {} } };", None),
         ("var foo = { get bar() { try { return a(); } catch {  } finally {  } } };", None),
+        (
+            "
+            var foo = {
+                get bar() {
+                    for (let i = 0; i<10; i++) {
+                        return i;
+                    }
+                }
+            }
+            ",
+            None,
+        ),
+        (
+            "
+            var foo = {
+                get bar() {
+                    let i = 0;
+                    while (i < 10) {
+                        return i;
+                    }
```

Comment:
Although this might actually return in all cases practically, in general it would be impossible to know if this actually returns or not (halting problem). So, we just assume that it won't and require it to be explicit

Reply:
Right, In most cases we can't detect it, However, We have `ControlFlowGraph::is_infinite_loop_start` which does a super simple constant evaluation that detects things like `while(true)` and `for (;;)`; We may want to enhance this rule in the future to pass things like this:

```js
class foo {
  get bar() {
    while(true) { return 0 }
  }
}
```

While linting code like:

```js
class foo {
  get bar() {
    while(false) { return 0 }
  }
}
```

However, it isn't necessary for now and seems out of the scope of this PR.

Reply:
You are right, We have means to detect infinite loops like `while(true)` and `for(;;)` via constant evaluation but supporting it here might be out of scope.